### PR TITLE
fe-8 implement post comment to an article

### PR DESCRIPTION
**Pull request by rangararazin**

### CODE REVIEW

File: src/component/CommentAdder.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { postComments } from "../utils/api";
+
+export const CommentAdder = ({ setAllComments }) => {
+  const { currentUser } = useContext(UserContext);
+  const [newComment, setNewComment] = useState("");
+  const { article_id } = useParams();
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    postComments(article_id, currentUser, newComment).then(( commentRes ) => {
```
Comment (callumu21) [inline src/component/CommentAdder.jsx:13]: You need a way of preventing users from pressing submit multiple times before this has completed. On a slow connection, it's currently possible to make multiple posts before the comment is reset and the initial post request is completed.

File: src/utils/api.js [hunk 2]
```diff
@@ -25,3 +25,15 @@ export const getComments = (article_id) => {
 export const patchVotes = (article_id, params) => {
   return myApi.patch(`/articles/${article_id}`, params).then((res) => {});
 };
+
+export const postComments = (article_id, user, body) => {
+  console.log(article_id, user, body);
```
Comment (callumu21) [inline src/utils/api.js:30]: This can be removed now.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Good start, Razin! I've left a few comments that need to be addressed before this can be approved. Once you've made the changes, please resubmit this PR 😄

File: src/component/CommentAdder.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        console.log(commentRes);
+      setAllComments((currComment) => {
+        return [commentRes, ...currComment];
+      });
+      setNewComment("");
+    });
+    alert("Your comment has been added !! ")
```
[the remark this answers is not in the document]
Comment (rangararazin) [reply src/component/CommentAdder.jsx:20]: Thanks Callum.Will do the changes and resubmit PR.

**Comment by cjplatten** [comment]
This is working ok now, please be careful when commiting changes to a branch as a lot of changes have been commited here that are not relevant to addressing the previous comments.

**Pull request merged**